src/components/Article/ViewSingleArticleComponent.jsx

```diff
@@ -1,29 +1,52 @@
 import React, { Component } from 'react';
-import {
-  Loader, Rating, Segment, Sidebar,
-} from 'semantic-ui-react';
+import { connect } from 'react-redux';
+import { Loader, Segment, Sidebar } from 'semantic-ui-react';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import SideBarMenu from '../Menu/Menu';
 import Header from '../Header/Header';
-
-
 import Footer from '../Footer/Footer';
+import { rateArticle } from '../../actions/ratingActions';
+import rateArticleComponent from './ratingArticleComponent';
 
 class ViewSingleArticleComponent extends Component {
+  // rating
+  constructor() {
+    super();
+    this.state = {
+      rating: 0,
+    };
+    this.onStarClick = this.onStarClick.bind(this);
+  }
+
   componentDidMount() {
     const { getArticle } = this.props;
     const { match } = this.props;
     getArticle(match.params.slug);
   }
 
+  // rating
+  onStarClick(nextValue) {
+    this.setState({ rating: nextValue });
+    const { match } = this.props;
+    const articleSlug = match.params.slug;
+    const payload = {
+      slug: articleSlug, rating: { rating: nextValue },
+    };
+    // eslint-disable-next-line no-shadow
```

Comment:
why are you disabling this?

Reply:
es lint was complaining: "Already defined in the upper scope", its import.